libs/libvtrutil/src/vtr_memory.h

```diff
@@ -46,6 +46,11 @@ void chunk_delete(T* obj, t_chunk* /*chunk_info*/) {
 //Cross platform wrapper around GNU's malloc_trim()
 // TODO: This is only used in one place within VPR, consider removing it
 int malloc_trim(size_t pad);
+
+// This is a macro because it has to be.  rw and locality must be constants,
```

Comment:
Thanks for wrapping in a macro.  I agree with Kevin it's important to minimize and modularize code like this that may not port or age well.

I'd also like a comment describing what it does.  Calls the x86 prefetch instruction to preload data into the cache, which should only be done for performance critical code where it shows a gain.  If in doubt, leave it to the HW prefetcher.

Reply:
Done